pkg/authorization/authorizer/accessrestriction/accessrestriction.go

```diff
@@ -0,0 +1,175 @@
+package accessrestriction
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/kubernetes/plugin/pkg/auth/authorizer/rbac"
+
+	userapiv1 "github.com/openshift/api/user/v1"
+	userlisterv1 "github.com/openshift/client-go/user/listers/user/v1"
+	"github.com/openshift/origin/pkg/authorization/apis/authorization"
+	authorizationlister "github.com/openshift/origin/pkg/authorization/generated/listers/authorization/internalversion"
+
+	"github.com/golang/glog"
+)
+
+func NewAuthorizer(accessRestrictionLister authorizationlister.AccessRestrictionLister, userLister userlisterv1.UserLister, groupLister userlisterv1.GroupLister) authorizer.Authorizer {
+	return &accessRestrictionAuthorizer{
+		accessRestrictionLister: accessRestrictionLister,
+		userLister:              userLister,
+		groupLister:             groupLister,
+	}
+}
+
+type accessRestrictionAuthorizer struct {
+	accessRestrictionLister authorizationlister.AccessRestrictionLister
+	userLister              userlisterv1.UserLister
+	groupLister             userlisterv1.GroupLister
+}
+
+func (a *accessRestrictionAuthorizer) Authorize(requestAttributes authorizer.Attributes) (authorizer.Decision, string, error) {
```

Comment:
this function is your meat and it needs to be re-written based on teh cycle conversation from a month plus ago.